File: 7 kyu/Number of People in the Bus/solution.clj
(ns kata.bus)
(defn number
  [bus-stops]
  (reduce - (apply map + bus-stops))
)
_____________________________________
(ns kata.bus)
(defn number
  [bus-stops]
  (->> bus-stops
    (map #(apply - %))
    (reduce +)))
_____________________________________
(ns kata.bus)
(defn number
  [bus-stops]
  (->> bus-stops
    (apply map +)
    (reduce -)))
_____________________________________
(ns kata.bus)

(defn delta [passengers [on off]]
  (+ passengers (- on off)))

(defn number
  [bus-stops]
  (reduce delta 0 bus-stops))
_____________________________________
(ns kata.bus)

(defn number
  [bus-stops]
  (reduce + (map (fn [e] (reduce - e)) bus-stops))
)

File: 7 kyu/Number of People in the Bus/solution.ex
defmodule Bus do
  def number([]), do: 0
  def number([{add, sub} | stops]) do
    add - sub + number(stops)
  end
end
_____________________________________
defmodule Bus do
  def number(stops) do
    Enum.reduce(stops, 0, fn({get_on, get_off}, total) ->
      total + get_on - get_off
    end)
  end
end
_____________________________________
defmodule Bus do

  @doc """
  Find the final number of people on the bus after given the amount of
  riders getting on and off the bus at each stop.
  """
  @spec number([{non_neg_integer, non_neg_integer}]) :: non_neg_integer
  def number(stops) do
    Enum.reduce(stops, 0, fn
      {a, b}, acc ->
        acc + a - b
    end)
  end
end
_____________________________________
defmodule Bus do
  def number(stops) do
    {on, off} = Enum.unzip(stops)
    Enum.sum(on) - Enum.sum(off)
  end
end
